### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by blaubaer**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Engity's Bifröst
+
+## Welcome
+
+Bifröst (spoken as "Bee-frest"), is a highly customizable SSH server with several way to authorize a user and where and how to execute its session. It can be used as a drop-in-replacement for [OpenSSH's sshd](https://man.openbsd.org/sshd), but it was actually created with some more advanced stuff in mind; see below.
```
Comment (blaubaer) [inline docs/index.md:5]: Test

File: docs/usecases.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+## On-board users within 15 minutes in the organization {:id=onboard}
+
+This is quite similar to [Off-board users within 15 minutes of the organization](#offboard), but obviously reverse.
+
+### Problem
+
+1. Assume you're part of an organization.
+2. Assume this organization has more than _just_ 10 people who might be able to access SSH resources.
+3. Assume you've to on-board an employee, now.
```
Comment (kzenon) [inline docs/usecases.md:62]: Assume you need to on-board an employee immediately.

File: docs/usecases.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+4. Assume you should ensure that this employee is ab to have access to all services, now.
```
Comment (kzenon) [inline docs/usecases.md:63]: Assume you have to ensure that this employee can access all services with no delay.

File: docs/usecases.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+In cases of SSH servers, this often results in running through all servers and either:
```
Comment (kzenon) [inline docs/usecases.md:65]: In case of SSH servers, this often results in going through all servers and either:

File: docs/usecases.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+* Share the server shared-user passwords,
+* Add user's public key to a shared user,
+* Add a dedicated user (with password or authorized key),
+* or change the [Ansible](https://www.ansible.com/) or [Puppet](https://www.puppet.com/) configuration apply it at every machine.
```
Comment (kzenon) [inline docs/usecases.md:70]: or changing the [Ansible](https://www.ansible.com/) or [Puppet](https://www.puppet.com/) configuration and apply it at every machine.

File: docs/usecases.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+How this should be done now (not days or weeks)?<br>
```
Comment (kzenon) [inline docs/usecases.md:72]: How can this be done quickly (not in days or weeks)?<br>

File: docs/usecases.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+"Did I really give him access everywhere?"
+
+### Solution
+
+Use the [OpenID Connect authorization](reference/authorization/oidc.md).
+
+There is no need to create them somewhere on the server itself. The [OIDC authorization](reference/authorization/oidc.md) will resolve them using the configured [Identity Provider (IdP)](https://openid.net/developers/how-connect-works/) - that's it!
```
Comment (kzenon) [inline docs/usecases.md:79]: There is no need to create them somewhere on the server itself. The [OIDC authorization](reference/authorization/oidc.md) will do that using the configured [Identity Provider (IdP)](https://openid.net/developers/how-connect-works/) - that's it!

File: docs/usecases.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+No need to access any of these services directly to create/authorize these users.
+
+If the [environments are configured accordingly](reference/environment/index.md) (which is the default) all user's resources (like home directory) will be created automatically.
+
+## Bastion Host / Jump Host {:id=bastion}
+
+### Problem
+
+1. Assume you've to manage resources.
```
Comment (kzenon) [inline docs/usecases.md:89]: Assume you have to manage resources.

File: docs/usecases.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+2. These resources are not directly reachable for you. They are protected inside other networks, you have no direct access to. For example, you're sitting at home and there is a another service inside an [AWS private VPC](https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html).
```
Comment (kzenon) [inline docs/usecases.md:90]: These resources are not directly accessible to you. They are protected within other networks to which you have no direct access. For example, you're sitting at home and there's another service inside an [AWS private VPC]. (https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html).

File: docs/usecases.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+3. You have to manage this service.
```
Comment (kzenon) [inline docs/usecases.md:91]: 3. You have to manage that service.

File: docs/usecases.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+   2. or outside with a fixed VPN connection inside the private network.
+2. Configure your favored [authorization](reference/authorization/index.md) (for example [OpenID Connect](reference/authorization/oidc.md) for best [on-boarding](#onboard) and [off-boarding](#offboard) experience).
+
+## Different rules for different users per host {:id=multi-environment}
+
+### Problem
+
+1. Assume you have an SSH server.
+2. Different users should be authorized differently.
+3. Different users should be executed into different [environments](reference/environment/index.md) (the one into a local one with permission A, another one with permission B and the other one into a remote one).
```
Comment (kzenon) [inline docs/usecases.md:111]: 3. Different users should run in different [environments](reference/environment/index.md) (one in a local environment with permission A, another with permission B, and a third user in a remote environment).

File: docs/usecases.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+This is currently not really possible, except with different [OpenSSH sshd](https://man.openbsd.org/sshd.8) setups on one host, or even different hosts or hacky [PAM](https://en.wikipedia.org/wiki/Linux_PAM) or [shell](https://en.wikipedia.org/wiki/Unix_shell) setups.
```
Comment (kzenon) [inline docs/usecases.md:113]: This is almost impossible with current technology, except with different [OpenSSH sshd](https://man.openbsd.org/sshd.8) setups on a host, or even different hosts, or hacked [PAM](https://en.wikipedia.org/wiki/Linux_PAM) or [shell](https://en.wikipedia.org/wiki/Unix_shell) setups.

File: docs/usecases.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+### Solution
+
+You're using Bifröst with multiple [flows](reference/flow.md), configured. Each flow can handle different authorization and environments.
```
Comment (kzenon) [inline docs/usecases.md:117]: Use Bifröst with multiple [flows](reference/flow.md), configured. Each flow can handle different authorizations and environments.

File: docs/reference/authorization/index.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
+---
+description: Everything in Bifröst starts here. A user is required to be authorized at first. Bifröst supports types, you can choose from to adjust to your needs.
+---
+
+# Authorizations
+
+Everything in Bifröst starts here. A user is required to be authorized at first. Bifröst supports [different types of authorizations](#types), you can choose from to adjust to your needs.
```
Comment (kzenon) [inline docs/reference/authorization/index.md:7]: Everything in Bifröst starts here. A user must be authorized at first. Bifröst supports [different types of authorizations](#types) you can choose from according to your needs.

File: docs/reference/templating/index.md [hunk 15]
```diff
@@ -0,0 +1,36 @@
+---
+description: In many places Bifröst can be customized using templating. How does they work?
+---
+
+# Templating
+
+In many places of the [configuration](../configuration.md) Bifröst can be not just customized using static strings or numbers, also templates can be used.
```
Comment (kzenon) [inline docs/reference/templating/index.md:7]: In many places of the [configuration](../configuration.md) Bifröst can not only be customized using static strings or numbers, but templates can be used.

File: docs/reference/templating/index.md [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+Bifröst is using the [template engine of Go](https://pkg.go.dev/text/template). Here is a collection of quite good resources to learn can be found at [HashiCrop's Nomand Developer documentation](https://developer.hashicorp.com/nomad/tutorials/templates/go-template-syntax).
```
Comment (kzenon) [inline docs/reference/templating/index.md:9]: Bifröst uses the [template engine of Go](https://pkg.go.dev/text/template). A collection of rather good resources to learn can be found at [HashiCrop's Nomand Developer documentation](https://developer.hashicorp.com/nomad/tutorials/templates/go-template-syntax).

File: docs/reference/connection/index.md [hunk 17]
```diff
@@ -0,0 +1,11 @@
+---
+description: How users are connecting to Bifröst and how to control them.
```
Comment (kzenon) [inline docs/reference/connection/index.md:2]: description: How users connect to Bifröst and how to control them.

File: docs/reference/connection/index.md [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+---
+
+# Connections
+
+Connections representing the way users connecting to Bifröst. Currently, Bifröst is supporting the [SSH protocol](ssh.md), only.
```
Comment (kzenon) [inline docs/reference/connection/index.md:7]: Connections represent the way users connect to Bifröst. Currently, Bifröst supports exclusively the [SSH protocol](ssh.md).

File: docs/reference/connection/ssh.md [hunk 19]
```diff
@@ -0,0 +1,88 @@
+---
+description: Defines the whole behavior of the SSH protocol by itself for a user who is connecting to Bifröst.
+---
+# SSH connection
+
+Defines the whole behavior of the SSH protocol by itself for a user who is connecting to Bifröst.
```
Comment (kzenon) [inline docs/reference/connection/ssh.md:6]: Defines the behavior of the SSH protocol for a user who is connecting to Bifröst.

File: docs/reference/authorization/local.md [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+| - | - |
+| `sshd` | _empty_ |
+
+## Password
+
+The password can either validated via `/etc/passwd` and `/etc/shadow` (default) or via PAM (if [`pamService`](#property-pamService) is set to a valid value).
+
+### Support of yescrypt {. #password-yescrypt}
+
+[yescrypt](https://en.wikipedia.org/wiki/Yescrypt) is cryptographic key derivation function used for password hashing in some modern Linux distributions (such as Ubuntu). Their support and give Bifröst the possibility to evaluate their passwords, the [`linux`/`extended` edition](../../setup/distribution.md#linux-extended) of Bifröst is required.
```
[the remark this answers is not in the document]
Comment (blaubaer) [reply docs/reference/authorization/local.md:38]: If yescrypt is not available Bifröst cannot validate password hashed in yescrypt. Do you have a better idea?

**Pull request merged**